### PR TITLE
Batch 7 of eslint error fixes - Closes #1535

### LINKED ISSUE #1535: Batch 7 of eslint error fixes
Child of #1485

### Which version(s) does this affect? (Environment, OS, etc...)
1.0.0

### PR DESCRIPTION
**Pull request by ManuGowda**
### What was the problem?
Disabled Eslint rules
```
block-scoped-var
default-case
no-console
no-restricted-properties
no-useless-escape
no-loop-func
no-multi-assign
```
### How did I fix it?
Enabled all the above-mentioned rules and fixed the errors
### Review checklist

* The PR solves #1535 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: helpers/bignum.js [hunk 1]
```diff
@@ -70,8 +70,8 @@ BigNumber.prototype.toBuffer = function(opts) {
 		}
 
 		var abs = this.abs();
-		var buf = abs.toBuffer({ size: 1, endian: 'big' });
-		var len = buf.length === 1 && buf[0] === 0 ? 0 : buf.length;
+		const buf = abs.toBuffer({ size: 1, endian: 'big' });
+		let len = buf.length === 1 && buf[0] === 0 ? 0 : buf.length;
```
Comment (willclarktech) [inline helpers/bignum.js]: It's strange to see `var`, `const`, and `let` together like this. How about declaring these as `var`s at the top of the function and reassigning both here and further down? Then at some point we can switch to `const` and `let` altogether if that makes sense.

Comment (willclarktech) [inline helpers/bignum.js]: It's strange to see `var`, `const`, and `let` together like this. How about declaring these as `var`s at the top of the function and reassigning both here and further down? Then at some point we can switch to `const` and `let` altogether if that makes sense.

File: helpers/constants.js [hunk 2]
```diff
@@ -81,6 +81,7 @@ var constants = {
 	},
 	feeStart: 1,
 	feeStartVolume: 10000 * 100000000,
+	// eslint-disable-next-line no-restricted-properties
```
Comment (willclarktech) [inline helpers/constants.js]: We should turn this off globally since we don't support ES2016.

File: helpers/http_api.js [hunk 3]
```diff
@@ -58,7 +58,7 @@ var middleware = {
 			return next();
 		}
 		logger.error(`API error ${req.url}`, err.message);
-		console.trace(err);
+		console.error(err);
```
Comment (willclarktech) [inline helpers/http_api.js]: This is quite different. We might want to add `.trace` to the list of allowed console methods.

File: helpers/http_api.js [hunk 4]
```diff
@@ -144,6 +144,7 @@ var middleware = {
 	 * @param {function} next
 	 */
 	applyAPIAccessRules(config, req, res, next) {
+		// eslint-disable-next-line no-useless-escape
```
Comment (willclarktech) [inline helpers/http_api.js]: Why is this necessary?

File: logic/round.js [hunk 5]
```diff
@@ -245,17 +245,19 @@ Round.prototype.applyRound = function() {
...
-				{
-					publicKey: delegate,
-					balance: self.scope.backwards ? -changes.balance : changes.balance,
-					u_balance: self.scope.backwards ? -changes.balance : changes.balance,
-					blockId: self.scope.block.id,
-					round: self.scope.round,
-					fees: self.scope.backwards ? -changes.fees : changes.fees,
-					rewards: self.scope.backwards ? -changes.rewards : changes.rewards,
-				},
+				accountData,
```
Comment (willclarktech) [inline logic/round.js]: Given the issue here is `no-loop-func`, this seems like an irrelevant change. Is this a bug in ESLint?

File: logic/transaction.js [hunk 6]
```diff
@@ -340,6 +340,7 @@ Transaction.prototype.checkBalance = function(
 					'Account does not have enough LSK:',
 					sender.address,
 					'balance:',
+					// eslint-disable-next-line no-restricted-properties
```
Comment (willclarktech) [inline logic/transaction.js]: As above.

File: modules/system.js [hunk 7]
```diff
@@ -220,6 +220,7 @@ System.prototype.versionCompatible = function(version) {
 	}
 
 	// if no range specifier is used for minVersion, check the complete version string (inclusive versionChar)
+	// eslint-disable-next-line no-useless-escape
```
Comment (willclarktech) [inline modules/system.js]: As above.

File: scripts/console.js [hunk 8]
```diff
@@ -94,5 +94,5 @@ process.on('uncaughtException', err => {
  * @listens unhandledRejection
  */
 process.on('unhandledRejection', (reason, p) => {
-	console.log('Unhandled Rejection at: Promise', p, 'reason:', reason);
+	console.error('Unhandled Rejection at: Promise', p, 'reason:', reason);
```
Comment (willclarktech) [inline scripts/console.js]: Should this be `error` or `warn`?

File: test/common/application.js [hunk 9]
```diff
@@ -66,6 +66,7 @@ function __init(initScope, done) {
 		// API: http://vitaly-t.github.io/pg-promise/global.html#event:extend
 		extend(object) {
 			Object.keys(dbRepos).forEach(repoName => {
+				// eslint-disable-next-line block-scoped-var
```
Comment (willclarktech) [inline test/common/application.js]: Can we get rid of this by declaring the `var` higher up?

File: test/common/modules_loader.js [hunk 10]
```diff
@@ -141,7 +141,8 @@ var modulesLoader = new function() {
 				new Logic(scope.logger, cb);
 				break;
 			default:
-				console.log('no Logic case initLogic');
+				console.info('no Logic case initLogic');
+				break;
```
Comment (willclarktech) [inline test/common/modules_loader.js]: Is break necessary?

File: test/functional/system/0.0.transfer.js [hunk 11]
```diff
@@ -28,6 +28,7 @@ describe('system test (type 0) - double transfers', () => {
 
 	var i = 0;
 	var t = 0;
+	/* eslint-disable no-loop-func */
```
Comment (willclarktech) [inline test/functional/system/0.0.transfer.js]: This while loop is very strange. Do we need it?

File: test/functional/system/2.2.delegates/1.same.account.same.username.js [hunk 12]
```diff
@@ -28,6 +28,7 @@ describe('system test (type 2) - double delegate registrations', () => {
 
 	var i = 0;
 	var t = 0;
+	/* eslint-disable no-loop-func */
 	while (i < 30) {
 		describe('executing 30 times', () => {
```
Comment (willclarktech) [inline test/functional/system/2.2.delegates/1.same.account.same.username.js]: This should have a different description each time at least. Maybe we need an issue to revisit these tests?

File: test/functional/system/0.0.transfer.js [hunk 13]
```diff
@@ -28,6 +28,7 @@ describe('system test (type 0) - double transfers', () => {
...
 	while (i < 1) {
 		describe('executing 30 times', () => {
```
Comment (willclarktech) [inline test/functional/system/0.0.transfer.js]: Description is inaccurate.

File: test/functional/ws/transport/transport.js [hunk 14]
```diff
@@ -239,6 +239,7 @@ describe('RPC', () => {
 
 		it('asking for a list multiple times should be ok', done => {
 			var successfulAsks = 0;
+			/* eslint-disable no-loop-func */
```
Comment (willclarktech) [inline test/functional/ws/transport/transport.js]: Let's talk about this one.

File: test/unit/helpers/http_api.js [hunk 15]
```diff
@@ -47,7 +47,7 @@ describe('httpApi', () => {
 				error: `API error: ${validError.message}`,
 			};
 			validNextSpy = sinonSandbox.spy();
-			spyConsoleTrace = sinonSandbox.spy(console, 'trace');
+			spyConsoleTrace = sinonSandbox.spy(console, 'error');
```
Comment (willclarktech) [inline test/unit/helpers/http_api.js]: As above.

File: logic/round.js (hunk 5 above)
Comment (ManuGowda) [reply logic/round.js]: yes this was a eslint error and after moving that a variable declaration the error went off.

File: test/functional/system/2.2.delegates/1.same.account.same.username.js (hunk 12 above)
Comment (ManuGowda) [reply test/functional/system/2.2.delegates/1.same.account.same.username.js]: There is actually a eslint rule for identical title, so we can club these issues along with that. This kind of description issues present across the project, so its a good idea to have a separate issue for this.

File: test/functional/ws/transport/transport.js (hunk 14 above)
Comment (ManuGowda) [reply test/functional/ws/transport/transport.js]: We discussed about this issue.

File: test/functional/system/0.0.transfer.js (hunk 11 above)
Comment (ManuGowda) [reply test/functional/system/0.0.transfer.js]: @willclarktech I think @diego-G will be able to answer this better, as I have no clue why we need to loop here for 30 times in specific or looping logic necessity.

Comment (diego-G) [reply test/functional/system/0.0.transfer.js]: These loops were useful to find bugs related to 0.9.11 & 0.9.12 patch. I suggest to keep them until porting 0.9.12.

File: test/functional/system/0.0.transfer.js (hunk 13 above)
Comment (ManuGowda) [reply test/functional/system/0.0.transfer.js]: As mentioned above.

File: test/functional/system/0.0.transfer.js (hunk 11 above)
Comment (willclarktech) [reply test/functional/system/0.0.transfer.js]: @diego-G The difference for this particular loop is it only runs once. Do you know why?

Comment (diego-G) [reply test/functional/system/0.0.transfer.js]: All those kind of loops from system test suite are doing the same thing through different type of transactions. I just wanted to maintain them there as a handy for development and testing. We can remove it later on.

File: .eslintrc.json [hunk 16]
```diff
@@ -64,6 +57,8 @@
 		"vars-on-top": "off",
 		"wrap-iife": "off",
 		// To keep
+		"no-restricted-properties": "off",
+		"no-console": ["error", { "allow": ["trace", "error", "info", "warn"] }],
```
Comment (willclarktech) [inline .eslintrc.json]: Prefer alphabetical in this section.

File: helpers/http_api.js [hunk 17]
```diff
@@ -144,7 +144,7 @@ var middleware = {
 	 * @param {function} next
 	 */
 	applyAPIAccessRules(config, req, res, next) {
-		if (req.url.match(/^\/peer[\/]?.*/)) {
+		if (req.url.match(/^\/peer[*]?.*/)) {
```
Comment (willclarktech) [inline helpers/http_api.js]: This doesn't seem equivalent.

File: scripts/console.js [hunk 18]
```diff
@@ -77,7 +77,7 @@ application.init(
  */
 process.on('uncaughtException', err => {
 	// Handle error safely
-	console.log('System error', { message: err.message, stack: err.stack });
+	console.warn('System error', { message: err.message, stack: err.stack });
```
Comment (willclarktech) [inline scripts/console.js]: Is `error` more appropriate here?

File: scripts/console.js (hunk 8 above)
Comment (willclarktech) [reply scripts/console.js]: @ManuGowda Thoughts on the above?

**Pull request merged**